### PR TITLE
Feat/book pages

### PR DESCRIPTION
**Pull request by eduardconstantin**
- [x] I have read and agree to the [Code of Conduct](https://github.com/danielschuster-muc/potter-db/blob/master/CODE_OF_CONDUCT.md), and have followed the [Contributing guidelines](https://github.com/danielschuster-muc/potter-db/blob/master/CONTRIBUTING.md).

## Pull Request Type

- [x] New feature (either for website or api)

## Summary

- Create book pages

### CODE REVIEW

**Review by danielschuster-muc — CHANGES_REQUESTED** [review]

File: frontend/src/components/pages/books/BookListItem.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            quality={30}
+            loading="eager"
+            priority
+          />
+        </CardMedia>
+        <CardHeader title={title} />
+        <CardContent>
+          <ListItemBox value={release_date} icon={faCalendar} />
+          <ListItemBox value={author} icon={faStarHalfStroke} />
+          <ListItemBox value={pages} icon={faClock} />
```
[the remark this answers is not in the document]
Comment (eduardconstantin) [reply frontend/src/components/pages/books/BookListItem.js:55]: Sure, I forgot about the icons

File: frontend/src/components/pages/index/SearchBar.js [hunk 2]
```diff
@@ -5,10 +5,14 @@ import { useState } from "react";
 
 const SearchBar = () => {
   const router = useRouter();
-  const [searchType, setSearchType] = useState("characters");
+  const [searchType, setSearchType] = useState("chapters");
```
Comment (danielschuster-muc) [inline frontend/src/components/pages/index/SearchBar.js:8]:
```suggestion
  const [searchType, setSearchType] = useState("characters");
```

**Review by danielschuster-muc — CHANGES_REQUESTED** [review]

File: frontend/src/lib/utils.js [hunk 3]
```diff
@@ -61,6 +61,16 @@ export const getPotionColorByDifficulty = (difficulty) => {
...
 };
 
+export const getBookColor = {
+  "harry-potter-and-the-philosopher-s-stone": "#140c36",
+  "harry-potter-and-the-chamber-of-secrets": "#518f4c",
+  "harry-potter-and-the-prisoner-of-azkaban": "#1b4993",
+  "harry-potter-and-the-goblet-of-fire": "#73aac9",
+  "harry-potter-and-the-order-of-the-phoenix": "#b43223",
+  "harry-potter-and-the-half-blood-prince": "#040907",
+  "harry-potter-and-the-deathly-hallows": "#51476a"
```
Comment (danielschuster-muc) [inline frontend/src/lib/utils.js:71]: Can you change the colors for the 1st and 6th book? Those are really hard to see. Maybe a bit lighter version of the 2 colors are better

File: frontend/src/lib/utils.js [hunk 4]
```diff
@@ -61,6 +61,16 @@ export const getPotionColorByDifficulty = (difficulty) => {
   }
 };
 
+export const getBookColor = {
```
Comment (danielschuster-muc) [inline frontend/src/lib/utils.js:64]:
```suggestion
export const getBookColorBySlug = {
```

File: frontend/src/lib/utils.js (hunk 3 above)
Comment (danielschuster-muc) [reply frontend/src/lib/utils.js:71]: And can you also put the whole method in a switch statement? Wwith a default color, if the slug doesn't match.

File: frontend/src/components/pages/books/BookList.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  return (
+    <>
+      <Typography variant="h3">Book Search</Typography>
+      <SearchField
+        placeholder="e.g. Philosopher's Stone"
+        handleChangeSearch={setSearchQuery}
+        totalResults={rawBooks?.pages
+            ? rawBooks?.meta?.pagination?.records
+            : 0}
```
Comment (danielschuster-muc) [inline frontend/src/components/pages/books/BookList.js:28]:
```suggestion
        totalResults={rawBooks?.meta?.pagination?.records || 0}
```
This should fix the results count.

File: frontend/src/lib/utils.js (hunk 3 above)
Comment (danielschuster-muc) [reply frontend/src/lib/utils.js:71]: And don't forget to change it in other places, too.

**Review by danielschuster-muc — CHANGES_REQUESTED** [review]
Just some small changes.

**Review by danielschuster-muc — APPROVED** [review]
Thank you for the changes, looks good now!

**Comment by eduardconstantin** [comment]
Great! You're welcome!

**Pull request merged**